src/unity/toolkits/supervised_learning/supervised_learning.cpp

```diff
@@ -1176,18 +1181,21 @@ variant_map_type supervised_learning_model_base::api_evaluate(
       std::string target = "class"; 
       std::string pred_column = "predicted_class";
 
-
       gl_sframe out;
 
       out["class"] = data[variant_get_ref<flexible_type>(state.at("target")).get<flex_string>()];
-      out["predicted_class"] = api_predict(data, missing_value_action_str, "class");
+      if(predictions.empty()) {
+        out[pred_column] = api_predict(data, missing_value_action_str, "class");
```

Comment:
This should use `target` rather than `"class"`, right?

Reply:
"class" is the type of prediction requested, not the target column.